### ISSUE TITLE
[core] DisableCursor() does not hide on web builds in windowed mode

### ISSUE THREAD

**Issue by grenappels** [raysan5/raylib#1895]
Please, before submitting a new issue verify and check:

 - [x] I tested it on latest raylib version from master branch
 - [x] I checked there is no similar issue already reported
 - [x] My code has no errors or misuse of raylib

### Issue description

This was working in raylib 3.5, which can be tested in the [first person example on the website](https://www.raylib.com/examples/web/core/loader.html?name=core_3d_camera_first_person). Expected behavior is that when the user clicks to focus on the canvas, it captures the cursor until the user presses escape. In 3.8 this does not happen. It looks like this is a result of removing the emscripten mouse callback that locked the pointer was [the solution to another user's problem here](https://github.com/raysan5/raylib/issues/1513). It *does* still work if you enter fullscreen mode; this issue is only for when the canvas is not fullscreen.

### Environment

Tested on Mac in Safari and Windows in Edge and Chrome

### Issue Screenshot

Here's my web build with 3.8 as an example: https://kyleappelgate.com/fire_guy/index.html -- I've also tried it separately with 3.5 which fixes the issue.

### Code Example

Just call `DisableCursor()` once before you start the game loop, it works on 3.5 but not on 3.8.

**Comment by raysan5** [comment]
@grenappels Actually I think the implementation is correct:
```c
// Enables cursor (unlock cursor)
void EnableCursor(void)
{
#if defined(PLATFORM_DESKTOP)
    glfwSetInputMode(CORE.Window.handle, GLFW_CURSOR, GLFW_CURSOR_NORMAL);
#endif
#if defined(PLATFORM_WEB)
    emscripten_exit_pointerlock();
#endif

    CORE.Input.Mouse.cursorHidden = false;
}

// Disables cursor (lock cursor)
void DisableCursor(void)
{
#if defined(PLATFORM_DESKTOP)
    glfwSetInputMode(CORE.Window.handle, GLFW_CURSOR, GLFW_CURSOR_DISABLED);
#endif
#if defined(PLATFORM_WEB)
    emscripten_request_pointerlock("#canvas", 1);
#endif

    CORE.Input.Mouse.cursorHidden = true;
}
```
but it could be [affected by web security](https://emscripten.org/docs/api_reference/html5.h.html#functions-affected-by-web-security).

**Comment by raysan5** [comment]
@grenappels Could you try activating that function on a mouse click event?
```c
if (IsMouseButtonPressed(MOUSE_BUTTON_LEFT)) DisableCursor();
```

**Comment by grenappels** [comment]
That doesn't work either, but I think I have found why the regression happened. In 3.5 there was a function `EmscriptenMouseCallback` and it was registered with `emscripten_set_click_callback("#canvas", NULL, 1, EmscriptenMouseCallback)`. Here's the funny thing: that function can be totally empty and just return 0, but adding it back to core.c makes DisableCursor() behave as it should. Meaning, you get no mouse locking if you don't ever call it, if you call it once at the start, the mouse locks the first time you click only, and if you call it on `IsMouseButtonPressed` it locks every time you click the window. Not sure if this is the right way to go about it but it unblocked me for now. I put the changes here if you wanna have a look: https://github.com/grenappels/raylib/commit/7be1d3f067224e4b59731dc76b71397557576312

**Comment by raysan5** [comment]
@grenappels Wow! This is very unexpected but, well, if it does the trick I can add that callback. Feel free to send a PR.

**Comment by grenappels** [comment]
done https://github.com/raysan5/raylib/pull/1900 😄
